datev_export_xml/views/templates.xml

```diff
@@ -32,7 +32,7 @@
 
         <accounting_info
             t-att-account_no="line.account_id.code or ''"
-            t-att-cost_category_id="line.analytic_account_id.code or ''"
+            t-att-cost_category_id="(line.analytic_line_ids[0].account_id.code or '') if line.analytic_line_ids else ''"
```

Comment:
This might be problematic if you actually use an analytic distribution or somebody manually set those (in different order) which would mean that sometimes an analytic account is used and sometimes a different one is used because the lines are ordered mainly by date and id.

Maybe a new `doc.datev_analytic_account(line)` function here? To allow an easier hook and custom selection of the "correct" analytic account?

Either way you could shorten it with `line.analytic_line_ids[:1].account_id.code or ''`